src/GitVersionCore/GitRepoInformation/Libgit2RepoMetadataProvider.cs

```diff
@@ -0,0 +1,87 @@
+﻿using LibGit2Sharp;
+using System.Collections.Generic;
+using System.Linq;
+using System;
+using System.Text.RegularExpressions;
+
+namespace GitVersion.GitRepoInformation
+{
+    public class Libgit2RepoMetadataProvider
+    {
+        public static GitRepoMetadata ReadMetadata(GitVersionContext context)
```

Comment:
Yeah I think so, just wasn't sure the shape at the time, this allowed me to just hook it in.

I am trying to make seams at the moment which will simplify code elsewhere, then can put the right abstraction in place